tests/events/events.go

```diff
@@ -0,0 +1,105 @@
+package events
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests/util"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type k8sObject interface {
+	metav1.Object
+	schema.ObjectKind
+}
+
+// ExpectNoEvent is safe to use in parallel as long as you are asserting namespaced object that is not shared between tests
+func ExpectNoEvent(object k8sObject, eventType, reason string) {
+	By("Expecting for an event to be not triggered")
+	expectEvent(object, eventType, reason, BeEmpty())
+}
+
+// ExpectEvent is safe to use in parallel as long as you are asserting namespaced object that is not shared between tests
+func ExpectEvent(object k8sObject, eventType, reason string) {
+	By("Expecting for an event to be triggered")
+	expectEvent(object, eventType, reason, Not(BeEmpty()))
+}
+
+// DeleteEvents is safe to use in parallel as long as you are asserting namespaced object that is not shared between tests
+func DeleteEvents(object k8sObject, eventType, reason string) {
+	By("Expecting events to be removed")
+	virtClient, err := kubecli.GetKubevirtClient()
+	util.PanicOnError(err)
+
+	fieldSelector, namespace := constructFieldSelectorAndNamespace(object, eventType, reason)
+
+	events, err := virtClient.CoreV1().Events(namespace).List(
+		context.Background(),
+		metav1.ListOptions{
+			FieldSelector: fieldSelector,
+		},
+	)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+	for _, event := range events.Items {
+		err = virtClient.CoreV1().Events(event.Namespace).Delete(context.Background(), event.Name, metav1.DeleteOptions{})
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
```

Comment:
Do you think it is worth attempting to delete all possible events and just let the error be defered in case the "no events exist" check also fails later? If the delete failed, but there are no events then we should be good, no?

---

The point of asserting in a library helper like this makes it hard to re-use the code in other places, or to allow the caller have more control of the information collected.
Some callers may decide to ignore it and continue, others may just want to copy the code to some other place which does not use ginkgo. When debugging, it is also easier to see the problem from the test itself and not deep in the calling stack. The use of the `*Offset` is many times problematic, just because we do not control from where it is called (e.g. not directly for a test body, but from some helper function).
I can share that I personally try to avoid using libs which use this internal assertion, for the above reasons.
Anyway, as this is a common pattern in this project, I would not insist beyond this message.

Reply:
Answering the first part, I am not 100% sure I got you but if there are no events(returned from the list) then we do nothing. If we fail to delete the event (for any reason other than not found) we need to fail otherwise next test can silently pass.

To the second point, yes generally it might be helpful to not assert(if the error is recoverable or one might want to ignore it). I agree with the `offset` I think I should differentiate between different asserts.  For example, don't hide internal errors and offset the others... 

The function is written this way because for now, we don't require anything else. We can extend or rework this if the need arises.

Reply:
Coming back to this. I am not sure what would be the benefit.